src/mscorlib/src/System/Collections/Generic/Comparer.cs

```diff
@@ -134,7 +132,6 @@ public override int Compare(T x, T y)
     // since we want to serialize as ObjectComparer for
     // back-compat reasons (see below).
 
-    [Serializable]
     internal sealed class Int32EnumComparer<T> : Comparer<T>, ISerializable where T : struct
```

Comment:
Do these need to remain serializable to make the `Comparer<T>` serialization work?